src/client/HostLobbyModal.ts

```diff
@@ -50,6 +51,18 @@ export class HostLobbyModal extends LitElement {
   private botsUpdateTimer: number | null = null;
   private userSettings: UserSettings = new UserSettings();
 
+  private mapLoaderResolve: (value: GameMapLoader) => void;
+  private mapLoader: Promise<GameMapLoader> = new Promise((resolve) => {
+    this.mapLoaderResolve = resolve;
+  });
+
+  /**
+   * @TODO: Get GameMapLoader from context
+   */
+  public injectMapLoader(mapLoader: GameMapLoader) {
+    this.mapLoaderResolve(mapLoader);
+  }
```

Comment:
Couldn't this be simplified?

```suggestion
  private mapLoader: GameMapLoader | undefined;

  public setMapLoader(mapLoader: GameMapLoader) {
    this.mapLoader = mapLoader;
  }
```